core/src/Modal.html

```diff
@@ -0,0 +1,134 @@
+<div class="fd-ui__overlay fd-overlay fd-overlay--modal" style="z-index:999">
+  <div class="fd-modal">
+    <div class="fd-modal__content" style="height:100%">
+      <div class="fd-modal__header">
+        <h1 class="fd-modal__title">{modalSettings.title}</h1>
+        <button
+          class="fd-button--light fd-modal__close"
+          on:click="fire('close')"
+          aria-label="close"
+        ></button>
+      </div>
+      <div class="iframeModalCtn"></div>
+    </div>
+  </div>
+</div>
+<script>
+  import * as Navigation from './navigation/services/navigation';
+  import * as RoutingHelpers from './utilities/helpers/routing-helpers';
+  import { LuigiConfig } from './services/config';
+  import * as GenericHelpers from './utilities/helpers/generic-helpers';
+
+  const prepareNodeData = async (component, path) => {
+    const pathUrlRaw =
+      path && path.length ? GenericHelpers.getPathWithoutHash(path) : '';
+    const pathData = await Navigation.getNavigationPath(
+      LuigiConfig.getConfigValueAsync('navigation.nodes'),
+      path
+    );
+    const params = RoutingHelpers.parseParams(pathUrlRaw.split('?')[1]);
+    const nodeParams = RoutingHelpers.getNodeParams(params);
+    const lastNode = RoutingHelpers.getLastNodeObject(pathData);
+    const modalSettings = component.get().modalSettings;
+    if (!modalSettings.title) {
+      modalSettings.title = lastNode.label;
+    }
+    component.get().isDataPrepared = true;
+    component.set({ modalSettings, lastNode, pathData, nodeParams });
+  };
+
+  const getNode = async (component, path) => {
+    if (component.get().isDataPrepared) {
+      const { nodeParams, lastNode, modalSettings, pathData } = component.get();
+      const iframe = createIframeModal(
+        lastNode.viewUrl,
+        {
+          context: pathData.context,
+          pathParams: pathData.pathParams,
+          nodeParams
+        },
+        component
+      );
+      component.root.set({
+        modalIframe: iframe,
+        modalIframeData: { ...pathData, nodeParams }
+      });
+    } else {
+      await prepareNodeData(component, path);
+    }
+  };
+  const createIframeModal = (viewUrl, componentData, component) => {
+    const elem = document.getElementsByClassName('fd-modal');
+    let sizeInPercent = '80%';
+    if (component.get().modalSettings.size) {
+      if (component.get().modalSettings.size === 'l') {
+        sizeInPercent = '80%';
+      } else if (component.get().modalSettings.size === 'm') {
+        sizeInPercent = '60%';
+      } else if (component.get().modalSettings.size === 's') {
+        sizeInPercent = '40%';
+      } else {
+        component.get().modalSettings['size'] = size;
+      }
+    }
+    if (!component.get().modalSettings.title) {
+      document.getElementsByClassName('fd-modal__title')[0].style.visibility =
```

Comment:
Please render it conditionally in template, not via CSS.

Reply:
I also thought about it, but this is a 'special' case. If no title is there e.g. for groups than we have two possibilities. Hide 'undefined' as modal__title or override css for fd-modal__title, because the height is broken.
In this case the conditional case is more pragmatic, for me.
![Bildschirmfoto 2019-04-01 um 15 00 58](https://user-images.githubusercontent.com/14272184/55329736-9f3ad400-548f-11e9-8515-ab9929a3cccb.png)

Reply:
You mean the title is `"undefined"` instead of `undefined`? If yes, I'd still vote for having `modal.title && modal.title!== 'undefined'` as a condition. If no, please explain it in an other way 😃